components/booking/AvailableTimes.tsx

```diff
@@ -1,112 +1,39 @@
-          times.splice(i, 1);
-        }
-
-        // Check if startTime is between slot
-        if (startTime.isBetween(dayjs(times[i]), dayjs(times[i]).add(props.eventType.length, "minutes"))) {
-          times.splice(i, 1);
-        }
-      });
-    }
-    // Display available times
-    setLoaded(true);
-  };
-
-  // Re-render only when invitee changes date
-  useEffect(() => {
-    setLoaded(false);
-    setError(false);
-    fetch(
-      `/api/availability/${user}?dateFrom=${props.date.startOf("day").utc().format()}&dateTo=${props.date
-        .endOf("day")
-        .utc()
-        .format()}`
-    )
-      .then((res) => res.json())
-      .then(handleAvailableSlots)
-      .catch((e) => {
-        console.error(e);
-        setError(true);
-      });
-  }, [props.date]);
-
+  const { rescheduleUid } = router.query;
+  const { slots, isFullyBooked, hasErrors } = Slots({ date, eventLength, workingHours });
   return (
     <div className="sm:pl-4 mt-8 sm:mt-0 text-center sm:w-1/3  md:max-h-97 overflow-y-auto">
       <div className="text-gray-600 font-light text-xl mb-4 text-left">
-        <span className="w-1/2">{props.date.format("dddd DD MMMM YYYY")}</span>
+        <span className="w-1/2">{date.format("dddd DD MMMM YYYY")}</span>
       </div>
-      {!error &&
-        loaded &&
-        times.length > 0 &&
-        times.map((time) => (
-          <div key={dayjs(time).utc().format()}>
+      {slots.length > 0 && (
+        slots.map((slot) => (
+          <div key={slot.format()}>
             <Link
               href={
-                `/${props.user.username}/book?date=${dayjs(time).utc().format()}&type=${props.eventType.id}` +
+                `/${user}/book?date=${slot.utc().format()}&type=${eventTypeId}` +
                 (rescheduleUid ? "&rescheduleUid=" + rescheduleUid : "")
               }>
-              <a
-                key={dayjs(time).format("hh:mma")}
-                className="block font-medium mb-4 text-blue-600 border border-blue-600 rounded hover:text-white hover:bg-blue-600 py-4">
-                {dayjs(time).tz(timeZone()).format(props.timeFormat)}
+              <a className="block font-medium mb-4 text-blue-600 border border-blue-600 rounded hover:text-white hover:bg-blue-600 py-4">
+                {slot.format(timeFormat)}
               </a>
             </Link>
           </div>
-        ))}
-      {!error && loaded && times.length == 0 && (
-        <div className="w-full h-full flex flex-col justify-center content-center items-center -mt-4">
-          <h1 className="text-xl font">{props.user.name} is all booked today.</h1>
-        </div>
+        ))
       )}
-      {!error && !loaded && <div className="loader" />}
-      {error && (
+      {isFullyBooked && <div className="w-full h-full flex flex-col justify-center content-center items-center -mt-4">
+        <h1 className="text-xl font">{user.name} is all booked today.</h1>
+      </div>}
```

Comment:
Codacy has a fix for the issue: [Replace `</div>` with `··</div>⏎······)`](https://app.codacy.com/gh/calendso/calendso/pullRequest?prid=7689306)
```suggestion
        </div>
```